pac4j-oauth/src/main/java/org/pac4j/oauth/profile/hiorgserver/HiOrgServerProfileDefinition.java

```diff
@@ -0,0 +1,84 @@
+package org.pac4j.oauth.profile.hiorgserver;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.github.scribejava.core.model.OAuth2AccessToken;
+import java.util.HashSet;
+import java.util.Set;
+import static org.pac4j.core.profile.AttributeLocation.PROFILE_ATTRIBUTE;
+import org.pac4j.core.profile.ProfileHelper;
+import org.pac4j.core.profile.converter.Converters;
+import org.pac4j.oauth.profile.JsonHelper;
+import org.pac4j.oauth.profile.definition.OAuth20ProfileDefinition;
+
+/**
+ * This class is the HiOrg-Server profile definition.
+ *
+ * @author Martin Böhmer
+ * @since 3.1.1
+ */
+public class HiOrgServerProfileDefinition extends OAuth20ProfileDefinition<HiOrgServerProfile, HiOrgServerConfiguration> {
+    
+    public static final String USER_ID = "user_id";
+    public static final String USERNAME = "username";
+    
+    public static final String NAME = "name";
+    public static final String FIRST_NAME = "vorname";
+    public static final String FULL_NAME = "fullname";
+    public static final String GROUP = "gruppe";
+    public static final String LEADER = "leitung";
+    public static final String POSITION = "funktion";
+    public static final String ORGANISATION_ID = "orga";
+    public static final String ORGANISATION_NAME = "organisation";
+    
+    protected static final String BASE_URL = "https://www.hiorg-server.de/api/oauth2/v1/user.php";
+    
+    public HiOrgServerProfileDefinition() {
+        super(x -> new HiOrgServerProfile());
+        primary(USERNAME, Converters.STRING);
+        primary(NAME, Converters.STRING);
+        primary(FIRST_NAME, Converters.STRING);
+        primary(FULL_NAME, Converters.STRING);
+        primary(GROUP, Converters.INTEGER);
+        primary(LEADER, Converters.BOOLEAN);
+        primary(POSITION, Converters.STRING);
+        primary(ORGANISATION_ID, Converters.STRING);
+        primary(ORGANISATION_NAME, Converters.STRING);
+    }
+    
+    @Override
+    public String getProfileUrl(OAuth2AccessToken accessToken, HiOrgServerConfiguration configuration) {
+        return BASE_URL;
+    }
+    
+    @Override
+    public HiOrgServerProfile extractUserProfile(String body) {
+        final HiOrgServerProfile profile = newProfile();
+        final JsonNode json = JsonHelper.getFirstNode(body);
+        if (json != null) {
+            logger.debug("Extracting user profile from JSON node " + json);
+            profile.setId(ProfileHelper.sanitizeIdentifier(profile, JsonHelper.getElement(json, USER_ID)));
+            for (final String attribute : getPrimaryAttributes()) {
+                convertAndAdd(profile, PROFILE_ATTRIBUTE, attribute, JsonHelper.getElement(json, attribute));
+            }
+        } else {
+            logger.debug("No JSON node to extract user profile was created");
```

Comment:
I don't think we want to build an empy profile, it seems to be a bad idea to me. Can you throw a `TechnicalException`instead?

Reply:
This is actually a behavior that all profiles have so far, i.e. none throws an exception. This is why I just added the log output. Throwing an exception makes a lot of sende, though. So, I agree to change that here, but it should be consitent across profiles.

Reply:
I'm in line. So let's change this one.

Added this issue for changing the others: https://github.com/pac4j/pac4j/issues/1175